File: src/main/java/amerebagatelle/github/io/afkpeace/AFKPeaceConfig.java
package amerebagatelle.github.io.afkpeace;

import org.jetbrains.annotations.Nullable;
import org.quiltmc.config.api.WrappedConfig;

public class AFKPeaceConfig extends WrappedConfig {
    // Setting toggles
    public final boolean autoAfk = false;
    public final boolean reconnectEnabled = false;
    public final boolean damageLogoutEnabled = false;
    public final boolean featuresEnabledIndicator = true;

    // Setting configurations
    public final int autoAfkTimerSeconds = 300;
    public final boolean reconnectOnDamageLogout = false;
    public final int secondsBetweenReconnectAttempts = 3;
    public final int reconnectAttemptNumber = 10;
    public final int damageLogoutTolerance = 20;
}

File: src/main/java/amerebagatelle/github/io/afkpeace/config/AFKPeaceConfig.kt
package amerebagatelle.github.io.afkpeace.config

import org.quiltmc.config.api.Config
import org.quiltmc.config.api.WrappedConfig

@Suppress("unused")
class AFKPeaceConfig : WrappedConfig() {
    val toggles = Toggles()
    val configurations: Configurations = Configurations()

    inner class Toggles : Config.Section {
        val autoAfk = false
        val reconnectEnabled = false
        val damageLogoutEnabled = false
        val featuresEnabledIndicator = true
    }

    inner class Configurations : Config.Section {
        val autoAfkTimerSeconds = 300
        val reconnectOnDamageLogout = false
        val secondsBetweenReconnectAttempts = 3
        val reconnectAttemptNumber = 10
        val damageLogoutTolerance = 20
    }
}